From 184fc357aeea71d0119b5d519892db106d1d736a Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 29 May 2019 20:15:47 +0000
Subject: [PATCH] TRITON-1709 Update "firewaller" unit tests to account for new
 "log" field Reviewed by: Pedro Palazon Candel <pedro@joyent.com> Approved by:
 Pedro Palazon Candel <pedro@joyent.com>

---
 Makefile             |  3 +++
 test/lib/rule.js     | 10 +++++++++-
 test/unit/helpers.js |  1 +
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ac34d1b..1c710cc 100644
--- a/Makefile
+++ b/Makefile
@@ -82,6 +82,9 @@ test: $(NODEUNIT_EXEC)
 		[[ $$? == "0" ]] || exit 1; \
 	done)
 
+$(NODEUNIT_EXEC): $(NPM_EXEC)
+	$(NPM) install
+
 .PHONY: release
 release: all docs $(SMF_MANIFESTS)
 	@echo "Building $(RELEASE_TARBALL)"
diff --git a/test/lib/rule.js b/test/lib/rule.js
index b0edd72..4a0b9f6 100644
--- a/test/lib/rule.js
+++ b/test/lib/rule.js
@@ -19,6 +19,14 @@ var h = require('../unit/helpers');
 var mocks = require('../unit/mocks');
 
 
+// --- Globals
+
+var AUTO_FILL = [
+    'created_by',
+    'log',
+    'version'
+];
+
 
 // --- Exports
 
@@ -64,7 +72,7 @@ function apiUpdateRule(t, opts, callback) {
         }
 
         if (opts.fillInMissing) {
-            [ 'created_by', 'version' ].forEach(function (f) {
+            AUTO_FILL.forEach(function (f) {
                 if (!opts.rule.hasOwnProperty(f)) {
                     opts.rule[f] = obj[f];
                 }
diff --git a/test/unit/helpers.js b/test/unit/helpers.js
index 40971e1..4e1232d 100644
--- a/test/unit/helpers.js
+++ b/test/unit/helpers.js
@@ -319,6 +319,7 @@ function equalSorted(t, actual, expected, desc) {
 function generateRule(override) {
     var rule = extend({
         enabled: true,
+        log: false,
         uuid: mod_uuid.v4(),
         version: '1'
     }, override);
-- 
2.21.0

